### PR TITLE
Misc components and effects for wagons.

### PR DESCRIPTION
**Pull request by WhistleWiz**
This includes the following:
* Wheel sparks
* Windows
* Shader replacer
* Internal/external switcher
* Distance optimiser

Plus some extra helper functions.

### CODE REVIEW

**Comment by WhistleWiz** [comment]
Hold this please, don't merge yet.

**Review by lassombra — COMMENTED** [review]
I'm going to make a couple minor changes to this before approving it - mostly related to the other comments

**Comment by lassombra** [comment]
> I'm going to make a couple minor changes to this before approving it - mostly related to the other comments

So it turns out I didn't consider that I don't have write access to your fork - so I think I'll go ahead and merge this and then make the couple of changes I am proposing on a new merge for you to look at.

**Review by lassombra — APPROVED** [review]

**Pull request merged**
